### ISSUE TITLE
[apex]AvoidGlobalModifierRule gives warning even when its a webservice.  False positive

### ISSUE THREAD

**Issue by pranayjswl007** [pmd/pmd#1348]
**Affects PMD Version:**  ALL

**Rule:**AvoidGlobalModifierRule

**Description:** When its a webservice , it will be declared as global. Even in that case it is causing warning. The rule should be smart enough to understand that its an exposed webservice and can be global

**Code Sample demonstrating the issue:**
 
```
@RestResource(urlMapping = '/partners/submit/*')


global class Generic_LoanCreation {
	@HttpPost
	global static void createLoanApplication() {
	
	}
}
```

```

```

**Running PMD through:** * Ant*

**Comment by adangel** [comment]
Thanks for the report.

Is it a requirement, that REST service classes need to be global? I've seen that the code samples in the Apex documentation, the classes are global there too, but I didn't see a clear statement, that they *must* be global. The same applies then for SOAP web services.

**Comment by JAertgeerts** [comment]
@adangel  

> To use this annotation, your Apex class must be defined as global.

https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_annotation_rest_resource.htm

**Comment by rsoesemann** [comment]
I think instead of changing the rule such an issue should be ignored when global must be used. The use of global always has risks (being unable to change the signature later, delete class from package,...). PMD for Apex is not a syntax checker it's a tool that warns developer to no misuse syntactically correct features of Apex.

**Comment by adangel** [comment]
For a Java rule, we usually add such exceptions to the rule, so that you don't need to configure suppressions/exclusions.

IMHO there is no other way than using global classes/methods, when implementing a webservice in Apex. That's why I consider this a valid false-positive.
If there is a way, we should change the description of the rule and point towards the alternative implementation for webservices.
